test-app/tests/integration/components/combobox-field-test.gts

```diff
@@ -0,0 +1,247 @@
+
+    assert.ok(
+      describedby.includes(hintId),
+      'Expected hintId to be included in the aria-describedby'
+    );
+  });
+
+  test('it renders with a hint and label block', async function (assert) {
+    await render(<template>
+      <ComboboxField data-combobox>
+        <:label><span data-label>label block content</span></:label>
+        <:hint><span data-hint>hint block content</span></:hint>
+      </ComboboxField>
+    </template>);
+
+    assert.dom('[data-hint]').hasText('hint block content');
+    assert.dom('[data-label]').hasText('label block content');
+  });
+
+  test('it renders with an error', async function (assert) {
+    await render(<template>
+      <ComboboxField @label="Label" @error="Error text" data-combobox />
+    </template>);
+
+    let error = find('[data-error]');
+
+    assert.dom(error).hasText('Error text');
+    assert.dom(error).hasAttribute('id');
+
+    let errorId = error?.getAttribute('id') || '';
+
+    assert.ok(errorId, 'Expected errorId to be truthy');
+
+    let describedby =
+      find('[data-combobox]')?.getAttribute('aria-describedby') || '';
+
+    assert.ok(
+      describedby.includes(errorId),
+      'Expected errorId to be included in the aria-describedby'
+    );
+
+    assert.dom('[data-combobox]').hasAttribute('aria-invalid', 'true');
+
+    assert.dom('[data-combobox]').hasClass('shadow-error-outline');
+    assert.dom('[data-combobox]').hasClass('focus:shadow-error-focus-outline');
+    assert.dom('[data-combobox]').doesNotHaveClass('shadow-focusable-outline');
+  });
+
+  test('it sets aria-describedby when both a hint and error are provided using the hint and errorIds', async function (assert) {
+    await render(<template>
+      <ComboboxField
+        @label="Label"
+        @error="Error text"
+        @hint="Hint text"
+        data-combobox
+      />
+    </template>);
+
+    let errorId = find('[data-error]')?.getAttribute('id') || '';
+
+    assert.ok(errorId, 'Expected errorId to be truthy');
+
+    let hintId = find('[data-hint]')?.getAttribute('id') || '';
+
+    assert.ok(hintId, 'Expected hintId to be truthy');
+
+    assert
+      .dom('[data-combobox]')
+      .hasAttribute('aria-describedby', `${errorId} ${hintId}`);
+  });
+
+  test('it disables the textarea using `@isDisabled` and renders a lock icon', async function (assert) {
+    await render(<template>
+      <ComboboxField @label="Label" @isDisabled={{true}} data-combobox />
+    </template>);
+
+    assert.dom('[data-combobox]').isDisabled();
+    assert.dom('[data-combobox]').hasClass('text-disabled');
+
+    assert.dom('[data-lock-icon]').exists();
+  });
+
+  test('it sets readonly on the textarea using `@isReadOnly` and renders a lock icon', async function (assert) {
+    await render(<template>
+      <ComboboxField @label="Label" @isReadOnly={{true}} data-combobox />
+    </template>);
+
+    assert.dom('[data-combobox]').hasAttribute('readonly');
```

Comment:
I personally think testing for the attribute is enough.  Since we are using the platform™, we can trust that the `readonly` attribute operates as defined at https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/readonly